### PR TITLE
Fix lint warnings from common components and flux stores

### PR DESCRIPTION
**Pull request by philschatz**
No UI Changes.

Fixes lint warnings from common components and flux stores.

### CODE REVIEW

File: src/flux/course.coffee [hunk 1]
```diff
@@ -46,7 +47,7 @@ CourseConfig =
     @emit('practice.loaded', obj.id)
 
   _reset: ->
-    CrudConfig.reset.call(@)
+    CrudConfig.reset.call(this)
```
Comment (pandafulmanda) [inline src/flux/course.coffee]: is explicitly saying `this` preferred over using `@` "standalone"/in these types of changes?

File: src/flux/task.coffee [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 _ = require 'underscore'
 flux = require 'flux-react'
 {TaskStepActions, TaskStepStore} = require './task-step'
 {CrudConfig, makeSimpleStore, extendConfig} = require './helpers'
 
 getSteps = (steps) ->
-  _.map(steps, ({id}) ->
+  _.map steps, ({id}) ->
     TaskStepStore.get(id)
-  )
```
Comment (pandafulmanda) [inline src/flux/task.coffee]: cool, so should we stop using unnecessary parens in current stuff as well?

**Comment by pandafulmanda** [comment]
just had some questions but :+1:

File: src/flux/course.coffee (hunk 1 above)
Comment (philschatz) [inline src/flux/course.coffee]: While it *does* find code like `that = @` I agree it's awkward. I'll probably just remove `no_stand_alone_at` from `coffeelint.json`

Comment (philschatz) [inline src/flux/course.coffee]: Original reason for discouraging empty `@`: https://github.com/jashkenas/coffeescript Issue 1601 (not making it linkable so we are not added to the issue)

Comment (pandafulmanda) [inline src/flux/course.coffee]: i see.  also good job on not accidentally linking, i totally did that on the react-calendar stuff lol.

File: src/flux/task.coffee (hunk 2 above)
Comment (philschatz) [inline src/flux/task.coffee]: I would say yes, but it's not enforceable... yet

**Pull request merged**